### PR TITLE
dwl: 0.3.1 -> 0.4

### PR DESCRIPTION
**Pull request by AndersonTorres**
###### Description of changes

https://github.com/NixOS/nixpkgs/pull/212462 triggered me

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/window-managers/dwl/default.nix
Comment (ckiee) [inline pkgs/applications/window-managers/dwl/default.nix:25]: This could probably go in `callPackage`.

File: pkgs/applications/window-managers/dwl/default.nix
Comment (ckiee) [inline pkgs/applications/window-managers/dwl/default.nix:27]:
```suggestion
stdenv.mkDerivation (final: {
```

**Review by ckiee — CHANGES_REQUESTED** [review]

File: pkgs/applications/window-managers/dwl/default.nix
Comment (AndersonTorres) [reply pkgs/applications/window-managers/dwl/default.nix:25]: Reference: https://github.com/NixOS/nixpkgs/issues/204303

File: pkgs/applications/window-managers/dwl/default.nix
Comment (AndersonTorres) [reply pkgs/applications/window-managers/dwl/default.nix:27]: Cosmetical, and I don't like the `final` word alone, isolated. final-what?

File: pkgs/applications/window-managers/dwl/default.nix
Comment (AndersonTorres) [reply pkgs/applications/window-managers/dwl/default.nix:68]: Thinking a little better, I will try to use make flags.

File: pkgs/applications/window-managers/dwl/default.nix
Comment (AndersonTorres) [reply pkgs/applications/window-managers/dwl/default.nix:103]: I am getting rid of `with`.

Comment (ckiee) [reply pkgs/applications/window-managers/dwl/default.nix:103]: Yes but what did `with` do to you here? It's not like there's going to be more stuff *inside the maintainers list*. It's just a list.

File: pkgs/applications/window-managers/dwl/default.nix
Comment (ckiee) [reply pkgs/applications/window-managers/dwl/default.nix:27]: My thinking was it would be a bit more like the `final: prev:` overlay pattern, which was comforting. rec -> fix is already pretty cosmetical.

Comment (AndersonTorres) [reply pkgs/applications/window-managers/dwl/default.nix:27]: Makes sense, I didn't think about this before.

However I don´t like these names for them, self-super was better.

File: pkgs/applications/window-managers/dwl/default.nix
Comment (AndersonTorres) [reply pkgs/applications/window-managers/dwl/default.nix]: Yes. I will reuse this.

File: pkgs/applications/window-managers/dwl/default.nix
Comment (AndersonTorres) [reply pkgs/applications/window-managers/dwl/default.nix:96]: If no one appears to share the maintainership, I can possibly undo this.

File: pkgs/applications/window-managers/dwl/default.nix
Comment (ckiee) [reply pkgs/applications/window-managers/dwl/default.nix:26]: `self:` is fine by me

File: pkgs/applications/window-managers/dwl/default.nix
Comment (ckiee) [reply pkgs/applications/window-managers/dwl/default.nix:96]: What harm comes from putting all of `lib.maintainers` into the list's inner scope?

File: pkgs/applications/window-managers/dwl/default.nix
Comment (AndersonTorres) [reply pkgs/applications/window-managers/dwl/default.nix:26]: I liked it!

File: pkgs/applications/window-managers/dwl/default.nix
Comment (AndersonTorres) [reply pkgs/applications/window-managers/dwl/default.nix:96]: In this particular case the readability is the same. 

----------------------
Specially I prefer 

```nix
builtins.attrValues {
  inherit (lib.maintainers) the various developers;
};
```

but no one besides me likes it.

File: pkgs/applications/window-managers/dwl/default.nix
Comment (ckiee) [inline pkgs/applications/window-managers/dwl/default.nix:59]: Why are we keeping this? We have `overrideAttrs`?

**Review by ckiee — CHANGES_REQUESTED** [review]

File: pkgs/applications/window-managers/dwl/default.nix
Comment (AndersonTorres) [reply pkgs/applications/window-managers/dwl/default.nix:59]: Better: I will remove it.

File: pkgs/applications/window-managers/dwl/default.nix
Comment (ckiee) [reply pkgs/applications/window-managers/dwl/default.nix:92]: I think this is fine since there is only one maintainer for now.

**Review by ckiee — APPROVED** [review]
I think this is failing successfully:
```
[nix-shell:~/.cache/nixpkgs-review/pr-212600]$ dwl
amdgpu: amdgpu_cs_ctx_create2 failed. (-13)
X11-1 title
X11-1 fullscreen
X11-1 floating
X11-1 selmon 0
X11-1 tags 0 1 0 0
X11-1 layout []=
X11-1 title
X11-1 fullscreen
X11-1 floating
X11-1 selmon 0
X11-1 tags 0 1 0 0
X11-1 layout []=
dwl: backend/x11/output.c:682: handle_x11_present_event: Assertion `buffer->n_busy > 0' failed.
Aborted (core dumped)
```
Nested Wayland-in-X11 is weird so I think this is still good to merge unless someone really wants to write a classic NixOS open-the-terminal-and-run-something test (:

**Comment by ckiee** [comment]
Result of `nixpkgs-review pr 212600` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dwl</li>
  </ul>
</details>

**Comment by AndersonTorres** [comment]
I remember a similar error in labwc. I reported upstream and it was indeed a bug.

**Pull request merged**
